// components/auth.js
var ee = require('vflow-pouchdb');
var page = require('page');
var h = require('vbridge').h;

function auth(state) {
  ee.on('session:active', function(e) {
    state.set('user', e.object.user);
    page.redirect('/');
  });

  ee.on('session:inactive', function(e) {
    state.set('user', null);
    page.redirect('/login');
  });

  setTimeout(function() {
    ee.emit('post', {
      verb: 'isactive',
      object: {
        type: 'session'
      },
      actor: null
    });
  }, 10);

  page('*', function(ctx, next) {
    if (ctx.path !== '/signup' 
      && ctx.path !== '/login' 
      && ctx.path !== '/sessions/create' 
      && ctx.path !== '/users/create') {
      if (!state.get('user')) return page.redirect('/login');
    }
    next();
  });

  page('/logout', function() {
    ee.emit('post', {
      verb: 'logout',
      object: {
        type: 'session',
      },
      actor: state.get('user')
    })
  });
}

function render(state) {
  return state.get('user') ? h('.u-pull-right', [
    h('span', state.get('user').name),
    h('span', ' | '),
    h('a', { href: '/logout' }, 'Logout')
  ]) : null;
} 

auth.render = render;
module.exports = auth;
