// components/show-profile.js
var ee = require('vflow-pouchdb');


var h = require('vbridge').h;
var page = require('page');

function profile(state) {
  ee.on('profile:get-show-response', function(doc) {
    state.set('profile', doc.object);
    state.set('ref', 'showProfile');
  });

  ee.on('profile:removed', function(doc) {
    page.redirect('/');
  });

  page('/profiles/:id/show', function(ctx) {
    ee.emit('post', {
      object: {
        type: 'profile',
        _id: ctx.params.id
      },
      verb: 'get-show',
      actor: state.get('user')
    });
    // show spinner?    
  });

  page('/profiles/:id/remove/:rev', function(ctx) {
    ee.emit('post', {
      object: {
        type: 'profile',
        _id: ctx.params.id,
        _rev: ctx.params.rev
      },
      verb: 'remove',
      actor: state.get('user')
    })
  });

}

function render(state) {
  var profile = state.get('profile');
  function listSkills() {
    if (!profile.skills) { return []; }
    return profile.skills.map(function(s) {
      return h('li', s);
    });
  }
  
  return h('div', [
    h('.row', [
      h('.twelve.columns', [
        h('.u-pull-right', [
          h('a', { href: 'http://twitter.com/' + profile.twitter}, '@' + profile.twitter),
          h('span', ' | '),
          h('a', { href: 'http://github.com/' + profile.github}, 'gh:' + profile.github)
        ]),
        h('h1', profile.name)
    
      ])    
    ]),
    h('.row', [
      h('.twelve.columns', [
        h('p', profile.email)
      ])
    ]),
    h('.row', [
      h('.twelve.columns', [
        h('p', profile.summary)
      ])
    ]),
    h('.row', [
      h('.twelve.columns', [
        h('h2', 'Skills'),
        h('ul', listSkills())
      ])
    ]),
    h('.row', [
      h('.twelve.columns', [
        h('ul', [
          h('li', [
            h('a', {href: '/profiles/' + profile._id + '/edit'}, ['Edit'])
          ]),
          h('li', [
            h('a', {href: '/profiles/' + profile._id + '/remove/' + profile._rev }, ['Remove'])
          ]),          
          h('li', [
            h('a', { href: '/'}, ['Return'])
          ])
        ])
      ])
    ])
  ])
}

profile.render = render;
module.exports = profile;
