// services/sessions.js
//var ee = require('vflow-pouchdb');
var ee = require('../../flow');

var pouchdb = require('pouchdb');
pouchdb.plugin(require('pouchdb-authentication'));

var db = pouchdb(window.location.origin + '/db');

module.exports = function() {
  function sync() {
    pouchdb.sync('devbase', window.location.origin + '/db', { live: true })
    .on('error', function(err) {
      console.log(err);
    });
  }

  ee.on('session:logout', function(doc) {
    db.logout(function(err, res) {
      ee.emit('post', {
        verb: 'inactive',
        object: {
          type: 'session',
        },
        actor: doc.actor,
        systemId: doc.systemId
      });
    });
  });

  ee.on('session:isactive', function(doc) {
    db.getSession(function(err, res) {
      if (err) { return console.log(err); }
      if (!res.userCtx.name ||  res.userCtx.name === 'root') {  
        ee.emit('post', {
          verb: 'inactive',
          object: {
            type: 'session'
          },
          actor: null,
          systemId: doc.systemId
        });
        return;
      }
      //sync();
      ee.emit('post', {
        verb: 'active',
        object: {
          type: 'session',
          user: { name: res.userCtx.name }
        },
        actor: { name: res.userCtx.name },
        systemId: doc.systemId
      });
    });
  });

  ee.on('session:create', function(doc) {
    // everyone can login
    db.login(doc.object.name, doc.object.password, function(err, res) {
      if (err) { 
        ee.emit('post', {
          verb: 'error',
          object: {
            type: 'session',
            message: err.message
          },
          systemId: doc.systemId
        });
        return console.log(err);
      }
      var e = {
        verb: 'created',
        object: {
          name: doc.object.name
        },
        systemId: doc.systemId
      };
      //sync();
      ee.emit('session:created', e);
    });
  });
};
